common/utils.go

```diff
@@ -84,3 +86,10 @@ func IsNil(i interface{}) bool {
 	}
 	return false
 }
+
+func LogOnPanicAndRethrow() {
```

Comment:
Though being specific is good, I think `LogOnPanic` is enough here. 
As a rule of thumb, the word `And` should not appear in the function name (in most cases). 

I understand your intention though, `LogOnPanic` might look like it will suppress the panic and only log it. 
Because it's basically a [decorator pattern](https://refactoring.guru/design-patterns/decorator), we could call this `DecoratePanic`, but that would be a bit not specific about the logging part 🤔